base_layer/transaction_components/src/key_manager/key_id.rs

```diff
@@ -191,6 +191,9 @@ impl FromStr for TariKeyId {
                     if parts.len() < 3 {
                         return Err("Wrong derived format".to_string());
                     };
+                    if id.contains("derived.managed.commitment mask.") {
+                        return Err("Wrong derived format".to_string());
+                    };
```

Comment:
```suggestion

```
we should not put special legacy stuff here

Reply:
you can change this to be more correct, and check that the second is actually a valid value. 
Derived key are `derived.X` where X = valid tari key id. 
So  in essence you can check that parts[1..].to_string = valid key id